### ISSUE TITLE
Extra gas consumption created by loops

### ISSUE THREAD

**Issue by bearcanrun** [Superalgos/ALGOToken#17]
In AlgoFees.mine(), `.length` of state variable `_miner` called in condition of _for_ loops. Every iteration of loop consumes extra gas.

*Note: a uint type declaration on `i` should also be made for AlgoFees.sol(L:121-129)*

**Code:**

```
function mine() public notTerminated onlySystem {
...

# AlgoFees.sol(L:81-101)
for(uint256 i = 1; i < _miners.length; i++) {

# AlgoFees.sol(L:121-129)
for(i = 1; i < _miners.length; i++) {

...
}
```

**Recommendation:**

Assign _miners.length to local variable

```
function mine() public notTerminated onlySystem {
...

// Add local variable
uint256 minerCount = _miners.length;

# AlgoFees.sol(L:81-101)
for(uint256 i = 1; i < minerCount; i++) {

# AlgoFees.sol(L:121-129)
for(uint256 i = 1; i < minerCount; i++) {
// Note 'i' type declaration added

...
}
```

**Comment by pmmax** [comment]
Fixed in #21

**Comment by bearcanrun** [comment]
Documentation updated.
